### PR TITLE
Adding variables parameter for query validations.

### PR DESCRIPTION
**Pull request by smotavitaendava**
Adding an extra interface for the internal validation.Validate for validate queries with GraphQL variables. Refactoring the existing `schema.Validate` function as well.

### CODE REVIEW

File: graphql.go [hunk 1]
```diff
@@ -144,14 +144,67 @@ type Response struct {
...
+		return []*errors.QueryError{errors.Errorf("%s", err)}, nil
+	}
+
+	if op.Type == query.Mutation {
+		if _, ok := s.schema.EntryPoints["mutation"]; !ok {
+			return []*errors.QueryError{{Message: "no mutations are offered by the schema"}}, nil
+		}
+	}
+
+	fields := []string{}
```
Comment (pavelnikolov) [inline graphql.go:168]: prefer default value, e.g. `var fields []string`

File: graphql.go [hunk 2]
```diff
@@ -144,14 +144,67 @@ type Response struct {
...
+	mainField := op.Selections[0].(query.Field)
+	for _, arg := range mainField.Arguments {
+		fieldName := arg.Name.Name + "."
+		fields = append(fields, fieldsFromLiteral(fieldName, arg.Value)...)
+	}
+
+	return nil, nil
```
Comment (pavelnikolov) [inline graphql.go:175]: You don't seem to be using the `fields` slice. Am I missing something? Why do you return `nil, nil`?

File: graphql.go [hunk 3]
```diff
@@ -144,14 +144,67 @@ type Response struct {
...
+}
+
+func fieldsFromLiteral(prefix string, lit common.Literal) []string {
+	fields := []string{}
```
Comment (pavelnikolov) [inline graphql.go:179]: `var fields []string`

**Comment by pavelnikolov** [comment]
Can you, please, add some unit test?

**Comment by smotavitaendava** [comment]
@pavelnikolov Thanks for the time you took reading this PR. Currently I'm working on this, I thought I've set the latest change (in the 'Adding field discovery' commit) out of this PR.
I'm looking forward to have 'Adding variables parameter for query validations.' revised.
Sorry any inconvenience.

**Pull request merged**